### PR TITLE
Tutorial/cloud storage infinite compose

### PR DESCRIPTION
**Pull request by domZippilli**
Conceptual overview of the APIs and appropriate programming techniques for being a power-user of the Cloud Storage compose feature.

### CODE REVIEW

**Comment by domZippilli** [comment]
fyi @ToddKopriva

**Comment by ToddKopriva** [comment]
I'll review this on Monday.

File: tutorials/cloud-storage-infinite-compose/index.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+```
+
+This approach trades off some complexity for a great increase in speed. In [this example command line tool](https://github.com/domZippilli/gcsfast/blob/master/gcsfast/cli/upload_standard.py), the technique is used to upload objects using parallel slices.
+
+
+## Conclusion
+
+Hopefully, this article has shown you how you can use the Cloud Storage compose function to create single objects from very large collections of other objects. It's a great thing to be able to do. I want to leave you with a few caveats:
+
+*   Remember that while there's no limit to the number of components, the 5TB object size limit does still apply.
```
Comment (anantdamle) [inline tutorials/cloud-storage-infinite-compose/index.md:291]: Although the number is very high.
Do you feel putting a number to [max number of components](https://cloud.google.com/storage/quotas#:~:text=Up%20to%2032%20objects%20can,composite%20object%20saturates%20at%202%2C147%2C483%2C647) (2,147,483,647) from a metadata perspective makes sense?

Comment (domZippilli) [reply tutorials/cloud-storage-infinite-compose/index.md:291]: I linked to that page on [line 28](https://github.com/GoogleCloudPlatform/community/pull/1531/files#diff-7c838efdb3715fb5f68b4a0c2bda8d2124ec27d4a2743ead2940babac0a5ad6eR28) and explained the integer saturation, so I think there's lots of information about this already.

Comment (ToddKopriva) [reply tutorials/cloud-storage-infinite-compose/index.md]: I think that we're well covered on this.

**Review by ToddKopriva — APPROVED** [review]

**Comment by ToddKopriva** [comment]
published:
https://cloud.google.com/community/tutorials/cloud-storage-infinite-compose

**Pull request merged**